src/MediatR/Mediator.cs

```diff
@@ -17,6 +17,9 @@ public class Mediator : IMediator
 
         private readonly MultiInstanceFactory _multiInstanceFactory;
 
+        private readonly Dictionary<Type, Type> genericHandlerCache;
+        private readonly Dictionary<Type, Type> wrapperHandlerCache;

```

Comment:
@jbogard What's the reason for using Profile259? Would it make sense to switch to `dotnet5.1`, which essentially is Profile259, but will work for future platforms as well.